x-pack/agentbeat/cmd/cmd.go

```diff
@@ -0,0 +1,45 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/v7/libbeat/cfgfile"
+	"github.com/elastic/beats/v7/libbeat/cmd"
+
+	"github.com/spf13/cobra"
+)
+
+func AgentBeat() *cobra.Command {
+	return prepareRootCommand()
+}
+
+func prepareCommand(rootCmd *cmd.BeatsRootCmd) *cobra.Command {
+	var origPersistentPreRun func(cmd *cobra.Command, args []string)
+	var origPersistentPreRunE func(cmd *cobra.Command, args []string) error
+	origPersistentPreRun = rootCmd.PersistentPreRun
+	origPersistentPreRunE = rootCmd.PersistentPreRunE
+	rootCmd.PersistentPreRun = nil
+	rootCmd.PersistentPreRunE = func(cmd *cobra.Command, args []string) error {
+		// same logic is used inside of *cobra.Command; if both are set the E version is used instead
```

Comment:
If it's the same, why are you replicating it here?

Reply:
Note: This is just a move of the content, this is not new code.

The comment is talking about the if/else statement. That code must be here because we are overriding the `PersistentPreRunE`, but then we add the `err := cfgfile.ChangeDefaultCfgfileFlag(rootCmd.Use)` below which is required for agentbeat to work.